### DISCUSSION TITLE
iconPos prop in the Button component not working properly

### DISCUSSION THREAD

**Discussion by mhyder1** [primefaces/community#2686] [PrimeReact]
iconPos should take one of four values: top, bottom, left and right. I can only get left and top to work. bottom is the same as top and right is the same as left. Is this the intended functionality? Thanks

**Comment by melloware** [comment]
@mhyder1 i don't see an iconPos property in the Menu?  https://primereact.org/menu/#api.MenuItem

Can you modify my StackBlitz and show me what you are doing?

https://stackblitz.com/edit/eun3bh?file=src%2FApp.tsx

**Reply by mhyder1** [reply]
My apologies. Its the Button component that takes that iconPos prop. I was using it with the menu component and go them confused. I'm using a font awesome icon with the Button. I updated the title and content of the original post.

**Reply by mhyder1** [reply]
I played with the stackblitz and I can get the icon placement of the button to work. but in my code im using FontAwesomeIcon as a prop. When I use FontAwesomeIcon the iconPos does not work. 
```
<Button
  icon={<FontAwesomeIcon icon={faCaretDown} />}
  iconPos="right"
/>
```
In this case iconPos stays on the left

**Reply by melloware** [reply]
Ahh the iconPos probaly add a CSS style to it that is probably not respected by FontAwesome?    If you can update the StackBlitz showing that I can take a look.

**Reply by mikechao** [reply]
I just encountered the same thing, but I am using Vue. Adding class="p-button-icon-right" did it for me.
```
      <Button
        type="submit"
        label="Send"
        icon-pos="right"
        :loading="isLoading"
        raised
        rounded
      >
        <template #icon>
          <font-awesome icon="fa-regular fa-paper-plane" class="p-button-icon-right" />
        </template>
      </Button>
```
